packages/react-api/src/api/model.js

```diff
@@ -66,15 +66,46 @@ export function executeModel(props) {
     filtersLogicalOperator
   };
 
-  // API supports multiple filters, we apply it only to geoColumn
-  const spatialFilters = spatialFilter
-    ? {
-        [source.geoColumn ? source.geoColumn : DEFAULT_GEO_COLUMN]: spatialFilter
+  let spatialFilters;
+  if (spatialFilter) {
+    let spatialDataType = source.spatialDataType;
+    let spatialDataColumn = source.spatialDataColumn;
+
+    if (!spatialDataType || !spatialDataColumn) {
+      if (source.geoColumn) {
+        const parsedGeoColumn = source.geoColumn ? source.geoColumn.split(':') : [];
+        if (parsedGeoColumn.length === 2) {
+          spatialDataType = parsedGeoColumn[0];
+          spatialDataColumn = parsedGeoColumn[1];
+        } else if (parsedGeoColumn.length === 1) {
+          spatialDataColumn = parsedGeoColumn[0] || DEFAULT_GEO_COLUMN;
+          spatialDataType = 'geo';
+        }
+        if (spatialDataType === 'geom') {
+          // fallback if for some reason someone provided old `geom:whatever`
+          spatialDataType = 'geo';
+        }
+      } else {
+        spatialDataType = 'geo';
+        spatialDataColumn = DEFAULT_GEO_COLUMN;
       }
-    : undefined;
+    }
+
+    // API supports multiple filters, we apply it only to geometry column or spatialDataColumn
+    spatialFilters = spatialFilter
+      ? {
+          [spatialDataColumn]: spatialFilter
+        }
+      : undefined;
 
-  if (spatialFilters) {
     queryParams.spatialFilters = JSON.stringify(spatialFilters);
+    queryParams.spatialDataType = spatialDataType;
+    if (spatialDataType !== 'geo') {
```

Comment:
I see `geo` repeated a few times in this file, so what do you think about extracting it somewhere and naming it as something like `const DEFAULT_SPATIAL_DATA_TYPE`?